App12/App12/Program.cs

```diff
@@ -0,0 +1,135 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+internal class Program
+{
+    static void Main(string[] args)
+    {
+        Console.WriteLine("Введите целое число m от 5 до 20");
+        int m;
+        if (!TryInputNumber(out m))
+        {
+            Console.ReadKey();
+            return;
+        }
+
+        Console.WriteLine("Введите целое число n от 5 до 20");
+        int n;
+        if (!TryInputNumber(out n))
+        {
+            Console.ReadKey();
+            return;
+        }
+
+        if (m < 5 || m > 20 || n < 5 || n > 20)
+        {
+            Console.WriteLine("Числа не удовлетворяют промежутку от 5 по 20");
+            Console.ReadKey();
+            return;
+        }
+
+        var array = new int[m, n];
+
+        var rnd = new Random();
+
+        for (int i = 0; i < array.GetLength(0); i++)
+            for (int j = 0; j < array.GetLength(1); j++)
+                array[i, j] = rnd.Next(100);
+
+        Console.WriteLine();
+        PrintArray(array);
+
+        int searchValue = ReadInt("\nВведите число для поиска в массиве: ", 0, 99);
+        Tuple<int, int> foundPosition = FindElement(array, searchValue);
+
+        if (foundPosition != null)
+        {
+            Console.WriteLine($"Элемент найден на позиции: строка - {foundPosition.Item1}, столбец - {foundPosition.Item2}");
```

Comment:
В терминологии есть ньюансы. В массивах индексация начинается с 0, а вот в таблицах строки / столбцы традиционно нумеруют с 1. То есть либо так
```suggestion
            Console.WriteLine($"Элемент найден на позиции: индекс строки - {foundPosition.Item1}, индекс столбца - {foundPosition.Item2}");
```
либо так
```suggestion
            Console.WriteLine($"Элемент найден на позиции: строка - {foundPosition.Item1 + 1}, столбец - {foundPosition.Item2 + 1}");
```